opencog/rule-engine/forwardchainer/ForwardChainer.h

```diff
@@ -48,39 +48,46 @@ class ForwardChainer {
 
     AtomSpace& _as;
     URECommons _rec;            // utility class
-	Handle _rbs;                // rule-based system
-	UREConfigReader _configReader;
-    FCMemory _fcmem;            // stores history
+    Handle _rbs;                // rule-based system
+    UREConfigReader _configReader;
     Logger * _log;
+
     int _iteration = 0;
     source_selection_mode _ts_mode;
+    bool _search_in_af;
+    bool _search_focus_Set;
+    Rule* _cur_rule;
+    Handle _cur_source;
+    HandleSeq _selected_sources;
 
-    void init();
-    void add_to_source_list(Handle h);
+    FCStat _fcstat;
 
-    void apply_all_rules(bool search_focus_set = false);
+    void init(Handle hsource, HandleSeq focus_set);
+    void setLogger(Logger* log);
+    Logger* getLogger(void);
 
+    void apply_all_rules(bool search_focus_set = false);
     void do_pm(const Handle& hsource, const UnorderedHandleSet& var_nodes);
```

Comment:
Yes, I removed it on my local branch since varialbe fulfillment queries should be handled by the backward chainer. Right?